config.go

```diff
@@ -1574,6 +1588,34 @@ func ValidateConfig(cfg Config, interceptor signal.Interceptor, fileParser,
 			maxChannelCommitInterval)
 	}
 
+	// Limit PendingCommitInterval so we don't wait too long for the remote
+	// party to send back a revoke.
+	if cfg.PendingCommitInterval > maxPendingCommitInterval {
+		return nil, mkErr("pending-commit-interval (%v) must be less "+
+			"than %v", cfg.PendingCommitInterval,
+			maxPendingCommitInterval)
+	}
+
+	// ChannelCommitInterval must be smaller than PendingCommitInterval,
+	// otherwise the channel link would fail before an HTLC is even
+	// attempted.
+	if cfg.ChannelCommitInterval >= cfg.PendingCommitInterval {
```

Comment:
Is this true? The pending commit ticker only starts once we've signed a commitment and the batch ticker (channel commit interval) only starts if we've got pending local updates to sign. They seem independent?

Reply:
I guess it depends on how fast the `localUpdateLog` updates. I wonder what would happen if the remote keeps sending updates without revoking the commit?

Basically when [this line](https://github.com/lightningnetwork/lnd/blob/master/htlcswitch/link.go#L1158) finishes, and there's no revc window, we'd start the loop again and now it depends on [the `PendingLocalUpdateCount`](https://github.com/lightningnetwork/lnd/blob/master/htlcswitch/link.go#L1085) I guess..

Reply:
> I wonder what would happen if the remote keeps sending updates without revoking the commit?

the local node would have to lock them in before doing anything except for settles, and would still fail the link

The link being failed shouldn't have anything to do with the batch ticker, since if the pending commit ticker is active, we can't sign for local updates anyways. Can you explain the comment/example in more detail?

Reply:
`localUpdateLog.logIndex`, incremented when process any update event `Add/Fail/Settle/MalformedFail/FeeUpdate`, via `appendHTLC` or `appendUpdate`.

`remoteCommit.ourMessageIndex`, updated only when the remote commit chain has unrevoked commitments when we have a `commitDiff` and via `lc.remoteCommitChain.addCommitment()`.

`ErrNoWindow`, returned when we have unacked remote commitment or do not have the next revocation point, which is handled in `ReceiveRevocation`, triggered when received `revoke_and_ack` from the remote.

Now my question is, can we have a remote commitment pending a `revoke_and_ack` from the remote, while the remote keeps sending updates such as `Add/Fail/Settle/MalformedFail/FeeUpdate`?

If so, this means,
- we'd have `PendingLocalUpdateCount() > 0`, so the batch ticker is resumed.
- we'd have `ErrNoWindow` from `SignNextCommitment()`, so the pending commit ticker is resumed.
Thus the two tickers are alive together.

A mental scenario is the remote doesn't send back a `revoke_and_ack` but keeps sending `update_fee`.

Reply:
> Now my question is, can we have a remote commitment pending a revoke_and_ack from the remote, while the remote keeps sending updates such as Add/Fail/Settle/MalformedFail/FeeUpdate?

yes

> we'd have PendingLocalUpdateCount() > 0, so the batch ticker is resumed.

these are for local updates, not remote

> Thus the two tickers are alive together.

I can't see why this matters? The only way the link fails is if the revoke_and_ack isn't sent, which has nothing to do with the batch ticker in this case?

Reply:
> these are for local updates, not remote

Not sure what you mean, but we create the remote commitment based on our local update logs?

1. say we don't have a revocation window, `PendingCommitTicker` resumed.
2. now we wait for the remote to send a revoke_and_ack to advance a previous state.
3. when the remote sends it, we'd rely on the `BatchTicker` to tick so we can do `l.updateCommitTxOrFail`.
4. since `BatchTicker` has a larger interval, the `PendingCommitTicker` would tick first.

Reply:
oh ok nvm I see this line `l.channel.OweCommitment(true)` inside `case *lnwire.RevokeAndAck:`, which does call `l.updateCommitTxOrFail` to reset the ticker. Will remove.